### DISCUSSION TITLE
select * from table 拉取表的全量数据，查询会因为内存超出限制失败

### DISCUSSION THREAD

**Discussion by lipenghao1216** [apache/doris#30108] [A - General / Q&A]
我有一个unique 模型的表t，数据量600多w行，数据解压后的大小在3g以上，我doris是一个fe3个be，都是单独部署的，be的可用内存平时一般有3g多。我单独运行select * from t 会因为内存使用超出限制而导致查询失败，通过memory tracker可以发现当时query消耗了3g多的内存导致be内存使用超过了水位线从取消了查询任务。
我认为如果只是单纯的select * from table，正常来讲不应该占用太多服务器的内存，数据库的通常做法不会是把所有数据都从硬盘load到内存中再返回给client，而是分批次的拉取返回给client（至少mysql我知道是这样的）。
有人遇到过类似问题吗，希望能有大神帮我解答一下，这是我的使用问题还是doris的bug。
我同样也提了一个issue，里面有更详细的信息
https://github.com/apache/doris/issues/30087

**Comment by LemonLiTree** [comment]
导入全表数据建议用export 或者 等2.1 版本的arrow sql

**Reply by lipenghao1216** [reply]
大神，感谢指点。其实我本身没有导出数据的需求，只是在对doris做各种测试的时候遇到了这个问题，对这个现象感到困惑，我对我的mysql库做了测试，运行select * from t 这种sql查询全表数据（表里数据量有两千多万条，1g多），服务器的内存使用基本没有变化（可能整个查询过程中会比平时多用几十m的内存），只要client自身有足够的内存就可以把数据全部拉取到。

**Reply by LemonLiTree** [reply]
doris跟mysql完全不一样。大数据查询输出对doris并不友好，所以我们开发了arrow sql 支持这种场景

**Comment by LemonLiTree** [comment]
可以参考这个issue 
https://github.com/apache/doris/issues/25514

**Reply by lipenghao1216** [reply]
好的，感谢
